src/Agent.Worker/Worker.cs

```diff
@@ -109,13 +111,22 @@ private void InitializeSecretMasker(Pipelines.AgentJobRequestMessage message)
             Trace.Entering();
             ArgUtil.NotNull(message, nameof(message));
             ArgUtil.NotNull(message.Resources, nameof(message.Resources));
-
             // Add mask hints for secret variables
             foreach (var variable in (message.Variables ?? new Dictionary<string, VariableValue>()))
             {
                 if (variable.Value.IsSecret)
                 {
                     HostContext.SecretMasker.AddValue(variable.Value.Value);
+                    // for variables, it is possible that they are used inside a shell which would strip off surrounding quotes
+                    // so, if the value is surrounded by quotes, add a quote-timmed version of the secret to our masker as well
+                    // This addresses issue #2525
+                    foreach (var quoteChar in _quoteLikeChars)
+                    {
+                        if (variable.Value.Value.StartsWith(quoteChar) && variable.Value.Value.EndsWith(quoteChar))
+                        {
+                            HostContext.SecretMasker.AddValue(variable.Value.Value.Trim(quoteChar));
```

Comment:
hm! does this trim both ends at once?

Reply:
According to: https://docs.microsoft.com/en-us/dotnet/api/system.string.trim?view=netcore-2.2
> Removes all leading and trailing occurrences of a set of characters specified in an array from the current string.